components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilder.java

```diff
@@ -260,8 +291,12 @@ public String buildIDToken(OAuthAuthzReqMessageContext authzReqMessageContext,
             return new PlainJWT(jwtClaimsSet).serialize();
         }
 
-        String signingTenantDomain = getSigningTenantDomain(authzReqMessageContext);
-        return OAuth2Util.signJWT(jwtClaimsSet, signatureAlgorithm, signingTenantDomain).serialize();
+        if (isIdTokenEncryptionEnabled(clientId)) {
+            return OAuth2Util.encryptJWT(jwtClaimsSet, encryptionAlgorithm, encryptionMethod, spTenantDomain, clientId).serialize();
+        } else {
+            String signingTenantDomain = getSigningTenantDomain(authzReqMessageContext);
+            return OAuth2Util.signJWT(jwtClaimsSet, signatureAlgorithm, signingTenantDomain).serialize();
```

Comment:
shall we make the return statement one line, as below?

return OAuth2Util.signJWT(jwtClaimsSet, signatureAlgorithm,getSigningTenantDomain(authzReqMessageContext)).serialize();